### PR TITLE
Add Evaluate Transforms FCurves node.

### PR DESCRIPTION
**Pull request by jekor**
Implements #1842 

Also add Frame output to Evaluate FCurve node.

### CODE REVIEW

File: animation_nodes/nodes/fcurve/evaluate_transforms_fcurves.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            yield "evaluationFrame += self.nodeTree.scene.frame_current_final"
+        yield "t = {'location': [0,0,0], 'rotation_euler': [0,0,0], 'scale': [1,1,1]}"
+        yield "outFrame = 0"
+        yield "for fcurve in fCurves:"
+        yield "    if fcurve.data_path in t:"
+        yield "        t[fcurve.data_path][fcurve.array_index] = fcurve.evaluate(evaluationFrame)"
+        yield "        outFrame = max(outFrame, fcurve.range()[1])"
+        yield "translation = Vector(tuple(t['location']))"
+        yield "rotation = Euler(tuple(t['rotation_euler']))"
+        yield "scale = Vector(tuple(t['scale']))"
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/fcurve/evaluate_transforms_fcurves.py:39]: `AN.utils.math.composeMatrix` doesn't really expect vectors, so you can just pass the lists for the scale and location directly, leaving the Euler as is.

File: animation_nodes/nodes/fcurve/evaluate_transforms_fcurves.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        self.newOutput("Float", "Frame", "outFrame")
+
+    def draw(self, layout):
+        layout.prop(self, "frameType", text = "Frame")
+
+    def getExecutionCode(self, required):
+        yield "evaluationFrame = frame"
+        if self.frameType == "OFFSET":
+            yield "evaluationFrame += self.nodeTree.scene.frame_current_final"
+        yield "t = {'location': [0,0,0], 'rotation_euler': [0,0,0], 'scale': [1,1,1]}"
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/fcurve/evaluate_transforms_fcurves.py:31]: Use a more meaningful name here. `transforms` maybe.

File: animation_nodes/nodes/fcurve/evaluate_transforms_fcurves.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import bpy
+from bpy.props import *
+from ... base_types import AnimationNode
+from ... events import executionCodeChanged
+
+frameTypeItems = [
+    ("OFFSET", "Offset", ""),
+    ("ABSOLUTE", "Absolute", "") ]
+
+class EvaluateFCurveNode(bpy.types.Node, AnimationNode):
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/fcurve/evaluate_transforms_fcurves.py:10]: Looks like you forgot to update the name of the class.

File: animation_nodes/nodes/fcurve/evaluate_transforms_fcurves.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    bl_idname = "an_EvaluateTransformsFCurvesNode"
+    bl_label = "Evaluate Transforms FCurves"
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/fcurve/evaluate_transforms_fcurves.py:12]: I would use "Evaluate FCurves Transforms" instead.

**Review by OmarEmaraDev — CHANGES_REQUESTED** [review]
Thanks for working on this!

I would leave the `outFrame` output out of this patch. I haven't looking at it, but it should be submitted as a different patch regardless.

Also, make sure to add your changes to the CHANGELOG file.

**Comment by jekor** [comment]
Updated.

File: animation_nodes/nodes/fcurve/evaluate_fcurves_transforms.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    ("OFFSET", "Offset", "", 0),
+    ("ABSOLUTE", "Absolute", "", 1) ]
+
+class EvaluateFCurvesTransformsNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_EvaluateFCurvesTransformsNode"
+    bl_label = "Evaluate FCurves Transforms"
+
+    frameType: EnumProperty(
+        name = "Frame Type", default = "OFFSET",
+        items = frameTypeItems, update = executionCodeChanged)
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/fcurve/evaluate_fcurves_transforms.py:18]: The update method should be `propertyChanged` now that we no longer change the execution code.

File: animation_nodes/nodes/fcurve/evaluate_fcurves_transforms.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        self.newOutput("Float", "Frame", "outFrame")
+
+    def draw(self, layout):
+        layout.prop(self, "frameType", text = "Frame")
+
+    def execute(self, fCurves, frame):
+        evaluationFrame = frame
+        if self.frameType == "OFFSET":
+            evaluationFrame += self.nodeTree.scene.frame_current_final
+        transforms = {"location": [0,0,0], "rotation_euler": [0,0,0], "scale": [1,1,1]}
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/fcurve/evaluate_fcurves_transforms.py:33]: Spaces after commas.

**Review by OmarEmaraDev — CHANGES_REQUESTED** [review]
Two small things and this will be ready to go.

**Review by OmarEmaraDev — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**